packages/runtime-core/src/helpers/useModel.ts

```diff
@@ -51,7 +51,10 @@ export function useModel(
       },
 
       set(value) {
-        if (!hasChanged(value, localValue)) {
+        if (
+          !hasChanged(value, localValue) &&
+          !(prevSetValue !== EMPTY_OBJ && hasChanged(value, prevSetValue))
```

Comment:
`prevSetValue` could be falsy, so we use `EMPTY_OBJ` as its initial value to check whether it has been set